dcos_test_utils/jobs.py

```diff
@@ -0,0 +1,169 @@
+""" Utilities for integration testing metronome in a deployed DC/OS cluster
+"""
+import logging
+
+import retrying
+from requests import HTTPError
+
+from dcos_test_utils.helpers import (ApiClientSession,
+                                     RetryCommonHttpErrorsMixin)
+
+REQUIRED_HEADERS = {'Accept': 'application/json, text/plain, */*'}
+log = logging.getLogger(__name__)
+
+
+class Jobs(RetryCommonHttpErrorsMixin, ApiClientSession):
+    def __init__(self, default_url, session=None):
+        super().__init__(default_url)
+        if session is not None:
+            self.session = session
+        self.session.headers.update(REQUIRED_HEADERS)
+        self._api_version = 'v1'
+
+    @staticmethod
+    def _run_from_history(run_id: str, history: dict) -> dict or None:
+        """Return a run item from a list of run items. This cas be
+        used with results from embedded history.
+
+        :param run_id: Run ID
+        :type run_id: str
+        :param history: History of Runs
+        :type history: dict
+        :return: Matching Run or None if not found
+        :rtype: dict or None
+
+        """
+        r = [run for run in history if run['id'] == run_id]
+        if r:
+            return r[0]
+        return None
+
+    def wait_for_run(self, job_id: str, run_id: str, timeout=600):
+        """Wait for a given run to complete or timeout seconds to
+        elapse.
+
+        :param job_id: Job ID
+        :type job_id: str
+        :param run_id: Run ID
+        :type run_id: str
+        :param timeout: Time in seconds to wait before giving up
+        :type timeout: int
+        :return: None
+
+        """
+
+        @retrying.retry(wait_fixed=1000, stop_max_delay=timeout * 1000,
+                        retry_on_result=lambda ret: ret is False,
+                        retry_on_exception=lambda x: False)
+        def _wait_for_run_completion(j_id: str, r_id: str) -> bool:
+            rc = self.get('{api}/jobs/{jid}/runs/{rid}'.format(
```

Comment:
In light of https://github.com/dcos/dcos-test-utils/pull/40 could we please make this `'/{api}/jobs/{jid}/runs/{rid}'` (note the leading slash) or just change self._api_version to be `/v1` instead of `v1`?